### PR TITLE
Change the recovery rule for 'new' to use expression_recovery.

### PR DESCRIPTION
**Pull request by EBatTiVo**
Fixes tivo/intellij-haxe #229
Added a unit test for the 'new' command.

### CODE REVIEW

**Comment by EBatTiVo** [comment]
@sganapavarapu1 @as3boyan, Can one of you please review this?  I'd like to have it in before Monday.

**Comment by as3boyan** [comment]
Seems fine to me. Approved for pull.

File: testData/parsing/haxe/expressions/New.hx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        return qqq;
+    }
+
+    public function getB(want_a:Bool):A {
+        var qqq = null != new B() ? new A() : new B();
+        return qqq;
+    }
+
+    public function getC(want_a:Bool):A {
+        var qqq = null != new B() ? want_a == true ? new A() : new B() : want_a == false ? new A() : new B();
```
Comment (as3boyan) [inline testData/parsing/haxe/expressions/New.hx]: test case looks a lil bit complex, but will do for now.

Comment (EBatTiVo) [inline testData/parsing/haxe/expressions/New.hx]: I agree.  I had tried a couple of combinations and found more errors after I fixed the first one.  So I decided to try ternaries embedded in ternaries, and what do you know?  I got errors.  So this is actually a useful test case.

**Pull request merged**